// @rescript/core/lib/js/src/Core__Nullable.js
// Generated by ReScript, PLEASE EDIT WITH CARE
'use strict';

var Caml_option = require("rescript/lib/js/caml_option.js");
var Core__Option = require("./Core__Option.js");

function fromOption(option) {
  if (option !== undefined) {
    return Caml_option.valFromOption(option);
  }
  
}

function equal(a, b, eq) {
  return Core__Option.equal((a == null) ? undefined : Caml_option.some(a), (b == null) ? undefined : Caml_option.some(b), eq);
}

function compare(a, b, cmp) {
  return Core__Option.compare((a == null) ? undefined : Caml_option.some(a), (b == null) ? undefined : Caml_option.some(b), cmp);
}

function getOr(value, $$default) {
  if (value == null) {
    return $$default;
  } else {
    return value;
  }
}

function getExn(value) {
  if (!(value == null)) {
    return value;
  }
  throw {
        RE_EXN_ID: "Invalid_argument",
        _1: "Nullable.getExn: value is null or undefined",
        Error: new Error()
      };
}

function forEach(value, f) {
  if (!(value == null)) {
    return f(value);
  }
  
}

function map(value, f) {
  if (value == null) {
    return value;
  } else {
    return f(value);
  }
}

function mapOr(value, $$default, f) {
  if (value == null) {
    return $$default;
  } else {
    return f(value);
  }
}

function flatMap(value, f) {
  if (value == null) {
    return value;
  } else {
    return f(value);
  }
}

var getWithDefault = getOr;

var mapWithDefault = mapOr;

exports.equal = equal;
exports.compare = compare;
exports.fromOption = fromOption;
exports.getOr = getOr;
exports.getWithDefault = getWithDefault;
exports.getExn = getExn;
exports.forEach = forEach;
exports.map = map;
exports.mapOr = mapOr;
exports.mapWithDefault = mapWithDefault;
exports.flatMap = flatMap;
/* No side effect */
